File: gerenciamento-cafe-api/src/main/java/com/luan/gerenciamentocafeapi/service/CategoriaService.java
package com.luan.gerenciamentocafeapi.service;

import com.luan.gerenciamentocafeapi.POJO.Categoria;
import org.springframework.http.ResponseEntity;

import java.util.List;
import java.util.Map;

public interface CategoriaService {
    ResponseEntity<String> addNewCategoria(Map<String, String> requestMap);

    ResponseEntity<List<Categoria>> getAllCategoria(String filterValue);

    ResponseEntity<String> updateCategoria(Map<String, String> requestMap);
}

File: gerenciamento-cafe-ui/src/app/services/categoria.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class CategoriaService {

  private url = `${environment.apiUrl}/categoria`; // Centralize a URL base

  private headers = new HttpHeaders({ 'Content-Type': 'application/json' }); // Defina os headers

  constructor(private httpClient: HttpClient) { }

  addNewCategoria(dados: any) {
    return this.httpClient.post(`${this.url}/add`, dados, { headers: this.headers });
  }

  updateCategoria(dados: any) {
    return this.httpClient.post(`${this.url}/update`, dados, { headers: this.headers });
  }

  getAllCategoria() {
    return this.httpClient.get(`${this.url}/get`);
  }

  // Método 'getFilterCategoria' que não aceita parâmetros.
  // Este método faz uma requisição GET para a URL '/get' com um parâmetro de query 'filterValue=true'
  // para obter categorias que atendam a um determinado filtro.
  getFilterCategoria() {
    return this.httpClient.get(`${this.url}/get?filterValue=true`);
  }
}
